pennylane/queuing.py

```diff
@@ -359,8 +354,9 @@ def append(self, obj, **kwargs):
         self[obj] = kwargs
 
     def remove(self, obj):
-        """Remove ``obj`` from the queue.  Raises ``KeyError`` if ``obj`` is not already in the queue."""
-        del self[obj]
+        """Remove ``obj`` from the queue. Passes silently if the object is not in the queue."""
+        if obj in self:
+            del self[obj]
```

Comment:
I feel like I asked this already but can't find the comment. What was the specific situation that required this change? I'm wondering if it's hiding something deeper.

Reply:
`update_info` first checks if the object is in the queue. Given that we changed all the `update_info` calls with calls to `remove`, I guess there is some part of the PL code that calls `update_info` with an object that might not be queued.

Reply:
Maybe we could rename it `remove_if_present`?

Reply:
ahh gotcha, sounds good. I think just `remove` is fine in this case, the lengthy name probably won't add much value